Idea使用swagger注解生成注释的时候报错 空指针

Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap.setPathParam, parameter value

[TRACE]	find config file:D:\833\gl-8.33\gl-app\.easy.api.config
[DEBUG]	use recommend config
[DEBUG]	use built-in config
[DEBUG]	load remote config
[TRACE]	find config file:D:\833\gl-8.33\gl-app\.easy.api.config
[DEBUG]	use recommend config
[DEBUG]	use built-in config
[DEBUG]	load remote config
[INFO]	search api from: com.ufgov.gl.app.gl.controller.rp.GlCurReceivePayController
[ERROR]	error eval script:
def maps = it.annMap("io.swagger.annotations.ApiImplicitParams")
for(map in maps.value){
def desc = map.value
def allowableValues = map.allowableValues
if(tool.notNullOrBlank(allowableValues)){
desc = desc + "n允许的值:" + allowableValues
}
def paramType = map.paramType ?: "query"
def required = map.required?:true
if(paramType==""||paramType=="query"){
api.setParam(map.name,map.defaultValue,required,desc)
}else if(paramType=="form"){
api.setFormParam(map.name,map.defaultValue,required,desc)
}else if(paramType=="path"){
api.setPathParam(map.name,map.defaultValue,desc)
}else if(paramType=="header"){
api.setHeader(map.name,map.defaultValue,required,desc)
}
}
[TRACE]	javax.script.ScriptException: javax.script.ScriptException: java.lang.NullPointerException: Parameter specified as non-null is null: method com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap.setPathParam, parameter value
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:158)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.eval(ScriptRuleParser.kt:97)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.access$eval(ScriptRuleParser.kt:34)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseEventRule$1.invoke(ScriptRuleParser.kt:77)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseEventRule$1.invoke(ScriptRuleParser.kt:76)
	at com.itangcent.intellij.config.rule.EventRule$Companion$of$1.compute(EventRule.kt:16)
	at com.itangcent.intellij.config.rule.EventRule$Companion$of$1.compute(EventRule.kt:14)
	at com.itangcent.intellij.config.rule.EventRule$filterWith$1.invoke(EventRule.kt:7)
	at com.itangcent.intellij.config.rule.EventRule$filterWith$1.invoke(EventRule.kt:5)
	at com.itangcent.intellij.config.rule.EventRule$Companion$of$1.compute(EventRule.kt:16)
	at com.itangcent.intellij.config.rule.EventRule$Companion$of$1.compute(EventRule.kt:14)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$RuleChainImpl.compute(DefaultRuleComputer.kt:74)
	at com.itangcent.intellij.config.rule.EventRuleMode$IGNORE_ERROR.compute(RuleMode.kt:78)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.doComputer(DefaultRuleComputer.kt:55)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:31)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:29)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:34)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.access$computerListener(RuleComputeListenerRegistry.kt:10)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:31)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:30)
	at com.itangcent.idea.utils.ContextualPsiClassHelper$InnerComputeListener.computer(ContextualPsiClassHelper.kt:261)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:30)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computer(RuleComputeListenerRegistry.kt:24)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:29)
	at com.itangcent.intellij.config.rule.RuleComputerKt.computer(RuleComputer.kt:35)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.processCompleted(RequestClassExporter.kt:322)
	at com.itangcent.idea.plugin.api.export.spring.SpringRequestClassExporter.processCompleted(SpringRequestClassExporter.kt:325)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.exportMethodApi(RequestClassExporter.kt:199)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.access$exportMethodApi(RequestClassExporter.kt:45)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:150)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:142)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:151)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:150)
	at com.itangcent.intellij.context.ActionContext$callInReadUI$1$1.invoke(ActionContext.kt:462)
	at com.itangcent.common.concurrent.ValueHolder.compute(Holder.kt:100)
	at com.itangcent.intellij.context.ActionContext.callInReadUI$lambda$15(ActionContext.kt:460)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:60)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:891)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:69)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:59)
	at com.itangcent.intellij.context.ActionContext.callInReadUI(ActionContext.kt:453)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:150)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:146)
	at com.itangcent.intellij.context.ActionContext.runAsync$lambda$8(ActionContext.kt:221)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.script.ScriptException: java.lang.NullPointerException: Parameter specified as non-null is null: method com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap.setPathParam, parameter value
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:320)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
	... 48 more
Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap.setPathParam, parameter value
	at com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap.setPathParam(RequestRuleWrap.kt)
	at com.itangcent.idea.plugin.api.export.rule.RequestRuleWrap$setPathParam.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:157)
	at Script277.run(Script277.groovy:16)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
	... 49 more

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/api/export/core/RequestBuilderListener.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/api/export/core/RequestBuilderListener.kt
@@ -201,7 +201,7 @@ fun RequestBuilderListener.addFormFileParam(
 
 fun RequestBuilderListener.addHeader(
     exportContext: ExportContext, request: Request,
-    name: String, value: String,
+    name: String, value: String?,
 ): Header {
     val header = Header()
     header.name = name
@@ -216,7 +216,7 @@ fun RequestBuilderListener.addHeader(
 
 fun RequestBuilderListener.addHeaderIfMissed(
     exportContext: ExportContext, request: Request,
-    name: String, value: String,
+    name: String, value: String?,
 ): Boolean {
     if (request.header(name) != null) {
         return false
@@ -230,7 +230,7 @@ fun RequestBuilderListener.addHeaderIfMissed(
 
 fun RequestBuilderListener.addPathParam(
     exportContext: ExportContext, request: Request,
-    name: String, desc: String,
+    name: String, desc: String?,
 ) {
     val pathParam = PathParam()
     pathParam.name = name
@@ -243,7 +243,7 @@ fun RequestBuilderListener.addPathParam(
 
 fun RequestBuilderListener.addPathParam(
     exportContext: ExportContext, request: Request,
-    name: String, value: String, desc: String?,
+    name: String, value: String?, desc: String?,
 ) {
     val pathParam = PathParam()
     pathParam.name = name
@@ -259,7 +259,7 @@ fun RequestBuilderListener.addResponseHeader(
     exportContext: ExportContext,
     response: Response,
     name: String,
-    value: String,
+    value: String?,
 ) {
     val header = Header()
     header.name = name

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/api/export/rule/RequestRuleWrap.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/api/export/rule/RequestRuleWrap.kt
@@ -266,11 +266,11 @@ class RequestRuleWrap(private val methodExportContext: MethodExportContext?, pri
         this.request.headers?.removeIf { it.name == name }
     }
 
-    fun addHeader(name: String, value: String) {
+    fun addHeader(name: String, value: String?) {
         requestBuilderListener.addHeader(methodExportContext!!, request, name, value)
     }
 
-    fun addHeaderIfMissed(name: String, value: String): Boolean {
+    fun addHeaderIfMissed(name: String, value: String?): Boolean {
         return requestBuilderListener.addHeaderIfMissed(methodExportContext!!, request, name, value)
     }
 
@@ -294,11 +294,11 @@ class RequestRuleWrap(private val methodExportContext: MethodExportContext?, pri
         }
     }
 
-    fun addPathParam(name: String, desc: String) {
+    fun addPathParam(name: String, desc: String?) {
         requestBuilderListener.addPathParam(methodExportContext!!, request, name, desc)
     }
 
-    fun addPathParam(name: String, value: String, desc: String) {
+    fun addPathParam(name: String, value: String?, desc: String?) {
         val pathParam = PathParam()
         pathParam.name = name
         pathParam.value = value
@@ -306,7 +306,7 @@ class RequestRuleWrap(private val methodExportContext: MethodExportContext?, pri
         requestBuilderListener.addPathParam(methodExportContext!!, request, pathParam)
     }
 
-    fun setPathParam(name: String, value: String, desc: String) {
+    fun setPathParam(name: String, value: String?, desc: String?) {
         val param = request.paths?.firstOrNull { it.name == name }
         if (param == null) {
             addPathParam(name, value, desc)

